### ISSUE TITLE
Replace Unit & fuel free-form strings with enums / calculated values in the form schemas

### ISSUE THREAD

**Issue by dleard** [bcgov/cas-reporting#346]
Because we were missing some clarity initially, we created our schemas with some free-form fields text for the units & fuels objects. We have the proper requirements / information around these now and can update those to be enum dropdowns / calculated non-editable fields where appropriate.

Todo:
 - Unit type should be dropdown (enum) of values as defined in the reporting fields spreadsheet. The enum can be defined directly in the source type schema as unit type selections have no determining relationships & do not need to be saved independently in the database. Where a set of unit types says "TBD" add an option for "Other". For example. GSC without line tracing has options in the spreadsheet of "Kiln", "Boiler", "TBD". The enum values for the unit type should be "Kiln", "Boiler", "Other".
 - Fuels need classifications {Woody Biomass, Other Exempted Biomass, Exempted Non-Biomass, Non-Biomass}. There is a tab in the Emission Summary spreadsheet that breaks the fuels up into groups.The fuels can have their classifications applied like so:
   - Fuels in Schedule C Item 1 are Woody Biomass
   - Fuels in Schedule C other than item 1 are Other Exempted Biomass
   - Fuels in Schedule C.1 are Exempted Non-Biomass
   - Fuels not covered by the 3 scenarios above are Non-Biomass
- Make an edit to the form_builder_service to fetch the list of all fuels and create the fuel_type enum from that list
- ~On selecting a fuel, the fuel classification should be automatically applied to the data as well~ Separate concern from the work in this ticket & has been split into #349 

__EDIT__:  It sounds like IRC is going to send us an exhaustive list of fuels along with their respective classifications. We should add the classifications from that list & check the list of fuels we currently have in the database with what is in the list we receive from IRC.

**Comment by dleard** [comment]
@patriciarussellCAS I remember you noted that some fields in the Units/Fuel sections didn't conform to the design (text fields where we should have dropdowns). We were waiting on some clarity around those fields before updating them, we now have that understanding & this ticket is to address that and make those changes from text fields to dropdowns with pre-defined options.

**Comment by dleard** [comment]
@patriciarussellCAS For reviewing this one: Any of the GSC activity pages will work, since they have both units & fuels. The visible change is that the unit type & fuel name fields are now dropdowns with pre-defined values to select from

**Comment by patriciarussellCAS** [comment]
My review:
- GSC Unit Type = dropdown confirmed
- Fuel Name = dropdown confirmed
- Fuel Classification = not confirmed. still not a dropdown for me in Dev. Is there a separate ticket for this? 
- Also note the bug I found in GSC non compression (#303) which might be related to this. 
@dleard

**Comment by dleard** [comment]
I split some of this work out into #349. Once that one is in, units & classification will be read-only fields that are populated when a fuel selection is made. It should also fix the bug in #303 as I've updated all the schemas & uiSchemas in #349 as well
